File: dotnet/src/xjustiz.core-dotnet/Util/Converter/ConversionResult.cs
namespace xjustiz.core_dotnet.Util.Converter;

using System.Collections.Generic;
using xjustiz.core_dotnet.Models;

/// <summary>
/// Result of the conversion process.
/// </summary>
public class ConversionResult
{
    /// <summary>
    /// Gets or sets the converted object.
    /// </summary>
    public UebermittlungSchriftgutobjekteNachricht? Result { get; set; }

    /// <summary>
    /// Gets or sets the list of data lost during conversion.
    /// </summary>
    public List<string> LostData { get; set; } = [];
}

File: sdk/java/example-api/src/main/java/de/xjustiz/example/api/util/ConversionResult.java
package de.xjustiz.example.api.util;

import de.xjustiz.example.api.models.UebermittlungSchriftgutobjekteNachricht;
import java.util.ArrayList;
import java.util.List;

public class ConversionResult {
    private UebermittlungSchriftgutobjekteNachricht result;
    private List<String> lostData = new ArrayList<>();

    public UebermittlungSchriftgutobjekteNachricht getResult() {
        return result;
    }

    public void setResult(UebermittlungSchriftgutobjekteNachricht result) {
        this.result = result;
    }

    public List<String> getLostData() {
        return lostData;
    }

    public void setLostData(List<String> lostData) {
        this.lostData = lostData;
    }
}
